pkg/preparer/setup.go

```diff
@@ -650,6 +652,71 @@ func getArtifactRegistry(preparerConfig *PreparerConfig) (artifact.Registry, err
 	return artifact.NewRegistry(url, fetcher, osversion.DefaultDetector), nil
 }
 
+func (p *Preparer) BuildRealityAtLaunch() error {
```

Comment:
this function would be nice to have a unit test for, you can set up a consul instance from a test with `consulutil.NewFixture()` and then seed it with whatever data you want, instantiate a preparer, run this function and check that reality has what you expect

Reply:
BuildRealityAtLaunch looks reads the filesystem in the path "/data/pods/*/current_manifest.yaml" to check what pods are installed on the disk. Is there a way to mock this in a test or should I create this file hierarchy on my local machine?

Reply:
This is resolved.